app/models/setting.rb

```diff
@@ -59,7 +59,7 @@ def self.per_page; 20 end # can't use our own settings
   def self.[](name)
     name = name.to_s
     cache_value = Setting.cache.read(name)
-    if cache_value.nil?
+    if cache_value.nil? && has_value?(name)
```

Comment:
interesting concept, but I don't know how. can I test that no query actually happened?